File: ecommerce_app/lib/kategori.dart
class Kategori {
  final int kategoriID;
  final String kategoriAdi;

  Kategori({
    required this.kategoriID,
    required this.kategoriAdi,
  });


  factory Kategori.fromJson(Map<String, dynamic> json) {
    return Kategori(
      kategoriID: json['kategoriID'] ?? 0,
      kategoriAdi: json['kategoriAdi'] ?? 'Kategori',
    );
  }
}

File: ECommerceBackEnd/ECommerceBackEnd/Models/Kategori.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace ECommerceBackEnd.Models
{
    public class Kategori
    {
        [Key]
        public int KategoriID { get; set; }

        [Column(TypeName = "VARCHAR")]
        [StringLength(30)]
        public string KategoriAdi { get; set; }

        [JsonIgnore]
        public ICollection<Urun>? uruns { get; set; }
    }
}
